src/include/session.h

```diff
@@ -96,6 +96,12 @@ struct __wt_session_impl {
 	size_t	  scratch_cached;	/* Scratch bytes cached */
 #ifdef HAVE_DIAGNOSTIC
 	/*
+	 * Variables used to look for violations of the contract that a
+	 * session is only used by a single session at once.
+	 */
+	volatile size_t api_tid;
+	volatile size_t api_enter_refcnt;
```

Comment:
The thread-ID should probably be `uintmax_t`, not `size_t`, and the reference count should be a `u_int`?

(Neither of them contain the size of an object in memory.)